tests/scalability/generate_brac_data/generate-brac-data.js

```diff
@@ -0,0 +1,221 @@
+    const assesmentFollowUpSurvey = bracSurvey.generateBracSurvey(
+      'assesment_follow_up', directParentPlace, place, person);
+    createDataDoc(reportsDirectory, assesmentFollowUpSurvey._id + dataExtension,
+      JSON.stringify(assesmentFollowUpSurvey, {}, 2));
+  }
+};
+
+const generateHierarchy = (type, placeName, numberOfPersons,
+  parentFirstLevelId, parentSecondLevelId, directParentPlace) => {
+  const place = bracPlaceFactory.generateBracPlace(placeName, type,
+    setParents(parentFirstLevelId, parentSecondLevelId));
+  let parentPersonThirdLevelId = null;
+  let parentPersonSecondLevelId = null;
+  let parentPersonFirstLevelId = null;
+  if (type === 'district_hospital') {
+    parentPersonFirstLevelId = place._id;
+  }
+  if (type === 'health_center') {
+    parentPersonFirstLevelId = place._id;
+    parentPersonSecondLevelId = parentFirstLevelId;
+    const randomIndex = Faker.faker.datatype.number(numberOfManagersPerDistrictHospital - 1);
+    place.supervisor = managers[randomIndex]._id;
+  }
+  if (type === 'clinic') {
+    parentPersonFirstLevelId = place._id;
+    parentPersonSecondLevelId = parentFirstLevelId;
+    parentPersonThirdLevelId = parentSecondLevelId;
+  }
+  let isPrimaryContact = true;
+  for (let i = 0; i < numberOfPersons; i++) {
+    const person = generatePerson(type, setParents(parentPersonFirstLevelId,
+      parentPersonSecondLevelId, parentPersonThirdLevelId), isPrimaryContact);
+    if (isPrimaryContact) {
+      place.contact = person;
+      createDataDoc(preconditionDataDirectory, place._id + dataExtension, JSON.stringify(place, {}, 2));
+    }
+    const needUser = pairPlaceTypesNeedsUsers[type];
+    if (needUser) {
+      generateUser(type, place._id, person.short_name.toLowerCase() + placeName + 'user' + i, person, isPrimaryContact);
+    }
+    if (bracPersonFactory.shouldGenerateSurvey(person)) {
+      generateReports(directParentPlace, place, person, (i % 2 !== 0));
+    }
+    isPrimaryContact = false;
+  }
+  return place;
+};
+
+const generateData = () => {
+  createDataDirectory(dataDirectory, preconditionDirectory);
+  createDataDirectory(dataDirectory, mainDirectory);
+  createDataDirectory(path.join(dataDirectory, preconditionDirectory), jsonDirectory);
+  for (let dh = 0; dh < numberOfDistrictHospitals; dh++) {
+    managers.splice(0, managers.length);
+    const districtHospital = generateHierarchy(
+      'district_hospital',
+      districtHospitalName + 'districthospital' + dh,
+      numberOfManagersPerDistrictHospital);
+
+    for (let hc = 0; hc < numberOfHealthCentersPerDistrictHospital; hc++) {
+      const healthCenter = generateHierarchy(
+        'health_center',
+        districtHospitalName + 'districthospital' + dh + 'healthcenter' + hc,
+        numberOfChwPerHealthCenter, districtHospital._id);
+
+      for (let c = 0; c < numberOfClinicsPerHealthCenter; c++) {
+        generateHierarchy('clinic',
```

Comment:
Please avoid inconsistent indentation, where some lines contain one argument, and other contain multiple arguments. It makes reading the code really difficult. Please, either inline all arguments, or one argument per line, like:

````javascript
function(arg1, arg2, arg3, arg4); 
function(   
  arg1, 
  arg2,
  arg3, 
  arg4
);
````

```suggestion
const clinicName = districtHospitalName + 'districthospital' + dh + 'healthcenter' + hc + 'clinic' + c;
await generateHierarchy('clinic', clinicName, numberOfFamilyMembers, healthCenter._id, districtHospital._id, healthCenter);

```